pkg/placement/selectors.go

```diff
@@ -72,3 +75,50 @@ func GetKeys(m map[string]string) []string {
 	}
 	return keys
 }
+
+// when an object is updated, we iterate over all placements and update placement-resolutions that
+// are affected by the update. Every affected placement decision is then queued to be synced.
+func (c *Controller) updateDecisions(obj runtime.Object) error {
+	placements, err := c.listPlacements()
+	if err != nil {
+		return err
+	}
+
+	objMR := obj.(mrObject)
+
+	for _, item := range placements {
+		placement, err := runtimeObjectToPlacement(item)
+		if err != nil {
+			return err
+		}
+
+		matchedSome := c.testObject(objMR, placement.Spec.Downsync)
+		if !matchedSome {
+			// if previously selected, remove
+			// TODO: optimize
+			if resolutionUpdated := c.placementResolver.RemoveObject(placement.GetName(), obj); resolutionUpdated {
+				// enqueue placement-decision to be synced since object was removed from its placement's resolution
+				c.enqueuePlacementDecision(placement.GetName())
+				c.logger.V(4).Info("enqueued placement-decision %s for syncing due to the removal of an " +
```

Comment:
FYI: I would use the type name PlacementDecision and I would identify the workload object.

Reply:
Oops, fixed.